tinodesdk/src/main/java/co/tinode/tinodesdk/Topic.java

```diff
@@ -855,8 +855,8 @@ private void processDelivery(final MsgServerCtrl ctrl, final long id) {
         }
     }
 
-    protected PromisedReply<ServerMessage> publish(final Drafty content, final long msgId) {
-        return mTinode.publish(getName(), content.isPlain() ? content.toString() : content).thenApply(
+    protected PromisedReply<ServerMessage> publish(final Drafty content, Map<String, Object> head, final long msgId) {
+        return mTinode.publish(getName(), content.isPlain() ? content.toString() : content, head).thenApply(
```

Comment:
This may create an inconsistency. If `Drafty` content was used to get headers, then `head` may contain `"mime"` header. So the client would send plain-text content with the header claiming it to be Drafty. I would delete "mime" value from the header, if header is not null.

Reply:
Added a check.